bot/seasons/evergreen/minesweeper.py

```diff
@@ -0,0 +1,233 @@
+
+        # Add game to list
+        board: GameDict = self.generate_board(bomb_chance)
+        revealed_board: GameDict = [["hidden"] * 10 for _ in range(10)]
+
+        await ctx.send(f"{ctx.author.mention} is playing Minesweeper")
+        chat_msg = await ctx.send(self.format_for_discord(revealed_board))
+
+        await ctx.author.send(
+            f"Play by typing: `{Client.prefix}reveal xy [xy]` or `{Client.prefix}flag xy [xy]` \n"
+            "Close the game with `.end`\n"
+            "Coordinates must be in format `<letter><number>`"
+        )
+        dm_msg = await ctx.author.send(self.format_for_discord(revealed_board))
+
+        self.games[ctx.author.id] = Game(
+            board=board,
+            revealed=revealed_board,
+            dm_msg=dm_msg,
+            chat_msg=chat_msg
+        )
+
+    async def update_boards(self, ctx: commands.Context) -> None:
+        """Update both playing boards."""
+        game = self.games[ctx.author.id]
+        await game.dm_msg.delete()
+        game.dm_msg = await ctx.author.send(self.format_for_discord(game.revealed))
+        await game.chat_msg.edit(content=self.format_for_discord(game.revealed))
+
+    @commands.dm_only()
+    @commands.command(name="flag")
+    async def flag_command(self, ctx: commands.Context, *Coordinates: CoordinateConverter) -> None:
+        """Place multiple flags on the board"""
+        board: GameDict = self.games[ctx.author.id].revealed
+        for x, y in Coordinates:
+            if board[y][x] == "hidden":
+                board[y][x] = "flag"
+
+        await self.update_boards(ctx)
+
+    async def lost(self, ctx: commands.Context) -> None:
+        """The player lost the game"""
+        game = self.games[ctx.author.id]
+        game.revealed = game.board
+        await ctx.author.send(":fire: You lost! :fire: ")
+        await game.chat_msg.channel.send(f":fire: {ctx.author.mention} just lost minesweeper :fire:")
+
+    async def won(self, ctx: commands.Context) -> None:
+        """The player won the game"""
+        game = self.games[ctx.author.id]
+        game.revealed = game.board
+        await ctx.author.send(":tada:  You won! :tada: ")
+        await game.chat_msg.channel.send(f":tada: {ctx.author.mention} just won minesweeper :tada:")
+
+    def reveal_zeros(self, revealed: GameDict, board: GameDict, x: int, y: int) -> None:
+        """Recursively reveal adjacent cells when a 0 cell is encountered."""
+        for x_, y_ in self.get_neighbours(x, y):
+            if revealed[y_][x_] != "hidden":
+                continue
+            revealed[y_][x_] = board[y_][x_]
+            if board[y_][x_] == 0:
+                self.reveal_zeros(revealed, board, x_, y_)
+
+    async def check_if_won(self, ctx, revealed: GameDict, board: GameDict) -> bool:
+        """Checks if a player has won"""
+        for x in range(10):
+            for y in range(10):
+                if revealed[y][x] == "hidden" and board[y][x] != "bomb":
```

Comment:
The second check seemed redundant since it would always be true if the first one is true.

```suggestion
                if revealed[y][x] == "hidden":
```

At this point you could consider using `any()` but the for loops are fine too.

Reply:
wait this would make it so you are required to flag all the spots to, I think I like that.

Reply:
That was unintended. I was actually incorrect because I didn't notice they're different boards (`revealed ` vs `board`)

Reply:
If you like this idea then maybe make a "hard mode" :) ?

Reply:
somebody mentioned that you actually need to mark with flags to win, but I don't know if that's correct.

Reply:
If i remember correctly you win both if you have all bombs flagged, or all non bombs opened.

Reply:
I think I will go with that you need to flag or reveal the spaces.

Reply:
on Wikipedia: `The game is won when all mine-free squares are revealed` so I think I will go with that.